sdkproject/Assets/Mapbox/Unity/MeshGeneration/LayerVisualizers/VectorLayerVisualizer.cs

```diff
@@ -143,55 +208,47 @@ public void SetProperties(VectorSubLayerProperties properties)
 					uvModOptions.atlasInfo = geometryMaterialOptions.atlasInfo;
 					uvModOptions.style = geometryMaterialOptions.style;
 
-					var uvMod = CreateInstance<UvModifier>();
+					var uvMod = AddOrCreateMeshModifier<UvModifier>();
 					uvMod.SetProperties(uvModOptions);
-					defaultMeshModifierStack.Add(uvMod);
 
 					if (_layerProperties.extrusionOptions.extrusionType != Map.ExtrusionType.None)
 					{
 						//replace materialOptions with styleOptions
 						if (geometryMaterialOptions.texturingType == UvMapType.Atlas || geometryMaterialOptions.texturingType == UvMapType.AtlasWithColorPalette)
 						{
-							var atlasMod = CreateInstance<TextureSideWallModifier>();
+							var atlasMod = AddOrCreateMeshModifier<TextureSideWallModifier>();
 							GeometryExtrusionWithAtlasOptions atlasOptions = new GeometryExtrusionWithAtlasOptions(_layerProperties.extrusionOptions, uvModOptions);
 							atlasMod.SetProperties(atlasOptions);
-							defaultMeshModifierStack.Add(atlasMod);
 						}
 						else
 						{
-							var heightMod = CreateInstance<HeightModifier>();
+							var heightMod = AddOrCreateMeshModifier<HeightModifier>();
 							heightMod.SetProperties(_layerProperties.extrusionOptions);
-							defaultMeshModifierStack.Add(heightMod);
 						}
 					}
 
 					//collider modifier options
 					if (_layerProperties.colliderOptions.colliderType != ColliderType.None)
 					{
-						var polyColliderMod = CreateInstance<ColliderModifier>();
+						var polyColliderMod = AddOrCreateGameObjectModifier<ColliderModifier>();
 						polyColliderMod.SetProperties(_layerProperties.colliderOptions);
-						defaultGOModifierStack.Add(polyColliderMod);
 					}
 
-					var styleMod = CreateInstance<MaterialModifier>();
-
+					var styleMod = AddOrCreateGameObjectModifier<MaterialModifier>();
 					styleMod.SetProperties(geometryMaterialOptions);
-					defaultGOModifierStack.Add(styleMod);
 
 					if (geometryMaterialOptions.texturingType == UvMapType.AtlasWithColorPalette)
 					{
-						var colorPaletteMod = CreateInstance<MapboxStylesColorModifier>();
+						var colorPaletteMod = AddOrCreateGameObjectModifier<MapboxStylesColorModifier>();
 						colorPaletteMod.m_scriptablePalette = geometryMaterialOptions.colorPalette;
-
-						defaultGOModifierStack.Add(colorPaletteMod);
 					}
 					break;
 				default:
 					break;
 			}
 
-			_defaultStack.MeshModifiers.AddRange(defaultMeshModifierStack);
-			_defaultStack.GoModifiers.AddRange(defaultGOModifierStack);
+			//_defaultStack.MeshModifiers.AddRange(defaultMeshModifierStack);
```

Comment:
Is this commented by mistake? Where are we adding default modifiers?

Reply:
@brnkhy

Reply:
@atripathi-mb it's adding default modifiers directly to the list. previously they were added to `defaultMeshModifierStack` and that list was added to the real mesh mod list using add range.